src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -226,9 +227,16 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                 // Write request line
                 await WriteStringAsync(request.Method.Method, cancellationToken).ConfigureAwait(false);
                 await WriteByteAsync((byte)' ', cancellationToken).ConfigureAwait(false);
-                await WriteStringAsync(
-                    _usingProxy ? request.RequestUri.AbsoluteUri : request.RequestUri.PathAndQuery,
-                    cancellationToken).ConfigureAwait(false);
+
+                if (_usingProxy)
+                {
+                    // Proxied requests contain full URL
+                    Debug.Assert(request.RequestUri.Scheme == Uri.UriSchemeHttp);
+                    await WriteBytesAsync(s_httpSchemeAndDelimiter, cancellationToken).ConfigureAwait(false);
+                    await WriteAsciiStringAsync(request.RequestUri.IdnHost, cancellationToken).ConfigureAwait(false);
```

Comment:
Should we use the Host header if present?

Reply:
I don't think so.  It's possible for the user to specify an IP address in the Uri and also specify an explicit Host header; the IP address is used to connect.  Presumably we want the same behavior for proxies, so we should send the host from the RequestUri.